### PR TITLE
Component

### PR DESCRIPTION
**Pull request by jamesgorrie**
Waiting on #61.
[A cleaner CR here](https://github.com/wellcometrust/wellcomecollection.org/compare/server-render-tidy...component?expand=1).

Adding the idea of a component tag to fit more inline with what we're doing on the pattern library.
Now if we make a navigation bar for example we can pass some arguments and make many of them.

It's a little better that macros by design by being more compartmentalised. You don't need a root like you do in macros, so each lives on their own.

e.g.
```
{% import input from "form-macro.njk" %}
{{ input("firstName", "text") }}

vs

{% component "form-input", { id: "firstName", type: "text" } %}{% endcomponent %}
```

I'm not sure how we add macros to the pattern lib, or if we'd want to.
I'm also not sure how the folder structure would work as we agreed on `component/{componantName}/index.njk`.

Seems this solves a few of those problems.

### CODE REVIEW

File: server/extensions/component.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import nunjucks from 'nunjucks';
+
+export default class Component {
+  constructor(env) {
+    this.tags = ['component'];
+    this.env = env;
+  }
+
+  parse(parser, nodes, lexer) {
```
Comment (davidpmccormick) [inline server/extensions/component.js]: what's `lexer` do?

**Comment by davidpmccormick** [comment]
Nice.

File: server/extensions/component.js (hunk 1 above)
Comment (jamesgorrie) [inline server/extensions/component.js]: It just get's passed in from the parser.

Comment (davidpmccormick) [inline server/extensions/component.js]: I'm easy – was just curious.

**Pull request merged**